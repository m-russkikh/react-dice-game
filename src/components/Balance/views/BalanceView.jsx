import React from 'react';
import {
  FormGroup,
  Label,
  Col,
  InputGroup,
  Input,
  InputGroupAddon,
  Button,
} from 'reactstrap';

import { BalanceViewPropType } from '../propTypes';


function BalanceView({ value, onButtonClick }) {
  return (
    <FormGroup row>
      <Label sm="3" md="2">Balance</Label>
      <Col>
        <InputGroup>
          <Input disabled value={value} />
          <InputGroupAddon addonType="append">
            <Button
              onClick={onButtonClick}
              disabled={value > 0}
            >
              Free Credits
            </Button>
          </InputGroupAddon>
        </InputGroup>
      </Col>
    </FormGroup>
  );
}

BalanceView.propTypes = BalanceViewPropType;

export default BalanceView;
