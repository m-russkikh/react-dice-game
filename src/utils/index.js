import { DEFAULT_BALANCE } from '../constants/balance';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getNextBetNumber() {
  return Math.floor(Math.random(0, 1) * 100) + 1;
}
