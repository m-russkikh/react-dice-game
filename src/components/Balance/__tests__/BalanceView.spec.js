import { getShallowWrapper } from '../../../utils/testUtils';
import BalanceView from '../views/BalanceView';


describe('BalanceView', () => {
  describe('balance value > 0', () => {
    let props;
    let wrapper;

    beforeAll(() => {
      props = {
        value: 10,
        onButtonClick: jest.fn(),
      };

      wrapper = getShallowWrapper(BalanceView, props);
    });

    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('balance value = 0', () => {
    let props;
    let wrapper;

    beforeAll(() => {
      props = {
        value: 0,
        onButtonClick: jest.fn(),
      };

      wrapper = getShallowWrapper(BalanceView, props);
    });

    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should handle button click', () => {
      wrapper.find('Button').simulate('click');

      expect(props.onButtonClick).toHaveBeenCalledTimes(1);
    });
  });
});
