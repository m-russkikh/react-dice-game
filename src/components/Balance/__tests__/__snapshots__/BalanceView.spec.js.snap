// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceView balance value = 0 should display correct 1`] = `
"<FormGroup row={true} tag=\\"div\\">
  <Label sm=\\"3\\" md=\\"2\\" tag=\\"label\\" widths={{...}}>
    Balance
  </Label>
  <Col tag=\\"div\\" widths={{...}}>
    <InputGroup tag=\\"div\\">
      <Input disabled={true} value={0} type=\\"text\\" />
      <InputGroupAddon addonType=\\"append\\" tag=\\"div\\">
        <Button onClick={[Function: mockConstructor]} disabled={false} color=\\"secondary\\" tag=\\"button\\">
          Free Credits
        </Button>
      </InputGroupAddon>
    </InputGroup>
  </Col>
</FormGroup>"
`;

exports[`BalanceView balance value > 0 should display correct 1`] = `
"<FormGroup row={true} tag=\\"div\\">
  <Label sm=\\"3\\" md=\\"2\\" tag=\\"label\\" widths={{...}}>
    Balance
  </Label>
  <Col tag=\\"div\\" widths={{...}}>
    <InputGroup tag=\\"div\\">
      <Input disabled={true} value={10} type=\\"text\\" />
      <InputGroupAddon addonType=\\"append\\" tag=\\"div\\">
        <Button onClick={[Function: mockConstructor]} disabled={true} color=\\"secondary\\" tag=\\"button\\">
          Free Credits
        </Button>
      </InputGroupAddon>
    </InputGroup>
  </Col>
</FormGroup>"
`;
